components/providers/RegionProvider.tsx

```diff
@@ -0,0 +1,24 @@
+"use client";
+import { DEFAULT_SELECTED_REGION } from "@/lib/rapid-hotel-api/constants/USER_OPTIONS";
+import { createContext, ReactNode, useState } from "react";
+
+export type RegionContextType = { region_id?: string; name?: string };
+
+export const RegionContext = createContext<
+  [RegionContextType, React.Dispatch<React.SetStateAction<RegionContextType>>]
+>([{}, () => {}]);
+
+const RegionProvider = ({ children }: { children: ReactNode }) => {
+  const [region, setRegion] = useState<RegionContextType>({
+    region_id: DEFAULT_SELECTED_REGION.region_id,
+    name: DEFAULT_SELECTED_REGION.regionNames.displayName
+  });
```

Comment:
```suggestion
  const [region, setRegion] = useState<RegionContextType>(() => {
    const storedRegion = localStorage.getItem("region");
    return storedRegion
      ? JSON.parse(storedRegion)
      : {
          region_id: DEFAULT_SELECTED_REGION.region_id,
          name: DEFAULT_SELECTED_REGION.regionNames.displayName,
        };
  });

  useEffect(() => {
    localStorage.setItem("region", JSON.stringify(region));
  }, [region]);
```

Store region data in `localStorage`, so its actually persistent.
![image](https://github.com/user-attachments/assets/5261604a-8d96-4c35-af4f-e363c6a93395)